programs/bpf_loader/src/syscalls/mod.rs

```diff
@@ -3108,6 +3113,151 @@ mod tests {
         call_program_address_common(seeds, address, &mut syscall)
     }
 
+    #[test]
+    fn test_syscall_sol_get_processed_sibling_instruction() {
+        let transaction_accounts = (0..8)
+            .map(|_| {
+                (
+                    Pubkey::new_unique(),
+                    AccountSharedData::new(0, 0, &bpf_loader::id()),
+                )
+            })
+            .collect::<Vec<_>>();
+        let mut transaction_context = TransactionContext::new(transaction_accounts, None, 3, 3);
+        for (index_in_trace, stack_height) in [1, 2, 1, 1, 2, 3, 2].into_iter().enumerate() {
```

Comment:
This test case looks good but the issue fixed by this PR is still untested. Could you add another test case like this?
```suggestion
        for (index_in_trace, stack_height) in [1, 2, 3, 2, 3, 4, 3].into_iter().enumerate() {
```

Reply:
I certainly can, but which scenario is not covered by the current `stack_height` sequence?

- We start at the end `index_in_trace`=6 and `stack_height`=2
- from there we can reach `index_in_trace`=4 and `stack_height`=2
- but are unable to reach `index_in_trace`=1 and `stack_height`=2
- In between there is a falling edge (going from `index_in_trace` 6 to 5) which we ignore
- a rising edge (going from `index_in_trace` 5 to 4) which we also ignore
- and finally the rising edge (going from `index_in_trace` 4 to 3) at which we `break`

Reply:
The current `stack_height` sequence still works without your PR changes because of the special handling for tx level instructions. The change in your PR for higher stack height handling is not being tested.

Reply:
I incremented the depth of the `stack_height` sequence as suggested

Reply:
Great, thanks!